src/app/rpmostree-clientlib.cxx

```diff
@@ -1682,7 +1685,31 @@ append_docs_to_str (GString          *str,
     }
 }
 
-/* Check whether sd_unit contains pid and return the boolean in sd_unit_contains_pid */
+/* Check whether sd_unit's `ActiveState` is `active` and return the boolean in `sd_unit_is_active`. */
+static gboolean
+check_sd_unit_state_is_active (char            *sd_unit,
+                               gboolean        *sd_unit_is_active,
+                               GDBusConnection *connection,
+                               GCancellable    *cancellable,
+                               GError         **error)
+{
+  g_autoptr(GVariant) active_state = NULL;
+  if (!get_sd_unit_property (connection, "LoadUnit", g_variant_new ("(s)", sd_unit),
+                             "ActiveState", &active_state, cancellable, error))
+    return FALSE;
+  else if (active_state)
+    {
+      const char *active_state_str =
+        g_variant_get_string (active_state, NULL);
+      if (g_strcmp0(active_state_str, "active") == 0)
```

Comment:
There are a few others that still technically count as "active" for our purposes (looking at https://www.freedesktop.org/wiki/Software/systemd/dbus/). Maybe something like:

```c
const char *active_states[] = {"active", "activating", "reloading", NULL};
*sd_unit_is_active = (active_state_str && g_strv_contains (active_states, active_state_str));
```

?

Reply:
On this topic, should we also add the `failed` state to the list? My intuition says we don't want to blindly ignore the driver if the machine usually relies on it but it crashed (i.e. it was meant to be enabled and is only not `active` because it crashed).

Reply:
double checked, it does :)